src/test/java/com/google/cloud/tools/maven/DeployAllMojoTest.java

```diff
@@ -0,0 +1,103 @@
+import org.apache.maven.plugin.MojoFailureException;
+import org.apache.maven.project.MavenProject;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+@RunWith(JUnitParamsRunner.class)
+public class DeployAllMojoTest {
+
+  @Rule public TemporaryFolder tempFolder = new TemporaryFolder();
+
+  @Mock private CloudSdkAppEngineFactory factoryMock;
+  @Mock private MavenProject project;
+  @Mock private AppEngineStandardStaging standardStagingMock;
+  @Mock private AppEngineDeployment deploymentMock;
+
+  @InjectMocks private DeployAllMojo deployAllMojo;
+
+  @Before
+  public void setup() throws IOException {
+    MockitoAnnotations.initMocks(this);
+    deployAllMojo.deployables = new ArrayList<>();
+    deployAllMojo.stagingDirectory = tempFolder.newFolder("staging");
+    deployAllMojo.sourceDirectory = tempFolder.newFolder("source");
+    when(project.getProperties()).thenReturn(new Properties());
+    when(project.getBasedir()).thenReturn(new File("/fake/project/base/dir"));
+    when(factoryMock.standardStaging()).thenReturn(standardStagingMock);
+    when(factoryMock.deployment()).thenReturn(deploymentMock);
+  }
+
+  @Test
+  @Parameters({"jar", "war"})
+  public void testExecute(String packaging)
+      throws IOException, MojoFailureException, MojoExecutionException, AppEngineException {
+    when(project.getPackaging()).thenReturn(packaging);
+
+    // create appengine-web.xml to mark it as standard environment
+    File appengineWebXml = new File(tempFolder.newFolder("source", "WEB-INF"), "appengine-web.xml");
+    appengineWebXml.createNewFile();
+    Files.write("<appengine-web-app></appengine-web-app>", appengineWebXml, Charsets.UTF_8);
+
+    // Make YAMLS
+    File appYaml = tempFolder.newFile("staging/app.yaml");
+    File cronYaml = tempFolder.newFile("staging/cron.yaml");
+    File dispatchYaml = tempFolder.newFile("staging/dispatch.yaml");
+    File dosYaml = tempFolder.newFile("staging/dos.yaml");
+    File indexYaml = tempFolder.newFile("staging/index.yaml");
+    File queueYaml = tempFolder.newFile("staging/queue.yaml");
+    File invalidYaml = tempFolder.newFile("staging/invalid.yaml");
+
+    deployAllMojo.execute();
+
+    assertTrue(deployAllMojo.deployables.contains(appYaml));
+    assertTrue(deployAllMojo.deployables.contains(cronYaml));
+    assertTrue(deployAllMojo.deployables.contains(dispatchYaml));
+    assertTrue(deployAllMojo.deployables.contains(dosYaml));
+    assertTrue(deployAllMojo.deployables.contains(indexYaml));
+    assertTrue(deployAllMojo.deployables.contains(queueYaml));
+    assertFalse(deployAllMojo.deployables.contains(invalidYaml));
+    verify(standardStagingMock).stageStandard(deployAllMojo);
+    verify(deploymentMock).deploy(deployAllMojo);
+  }
+}
```

Comment:
can you add a test to make sure it doesn't try to deploy a valid yaml that isn't actually in the directory?